databases/week2/updated_task_table.sql

```diff
@@ -0,0 +1,169 @@
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (31, 'Backup databases to external disk', 'Remember to store the disk in another physical location', '2017-09-09 17:32:33', '2017-10-01 21:18:59', '2017-12-23 14:21:01', 1, 2);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (32, 'Put up the new lamp in the hallway', NULL, '2017-10-15 05:45:54', '2017-10-16 14:05:35', '2017-12-29 02:29:26', 3, 3);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (33, 'Hang up paintings in living room', NULL, '2017-09-10 05:36:11', '2017-10-09 17:40:42', null, 3, 4);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (34, 'Buy plane ticket to Auckland', 'Check prices online first!', '2017-09-05 09:07:22', '2017-10-15 09:36:06', '2017-12-07 11:10:05', 1, 9);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (35, 'Learn about NoSQL databases', 'MongoDB, CouchDB, etc.', '2017-10-20 01:41:53', '2017-10-04 07:19:56', '2017-12-23 10:13:42', 2, NULL);
+select * from task;
+-- How many taska are in the table
+select count(id) from task;
+
+-- how many task in the task table dont have valid due date
+select count(title) from task where due_date is NULL;
+
+
+-- Find the all task that are marked as done
+select * from task where status_id = 3;
+
+-- find the tasks that are not marked as done
+select *from task where status_id <>3;
+
+-- Get all the task, sorted with the most recently created first.alter.
+select *from task order by created desc;
+
+-- get the single most recently created task
+-- select id, title, description, created, updated, due_date, status_id, user_id from task order by created desc limit 1;
+select * from task order by created desc limit 1;
+
+-- get the title and due date of all tasks where title or description contains database.
+select title, due_date from task where title  like '%database%' or description like '%database%';
+
+-- get the title and status (as text) of all tasks. 
+select  task.title, status.name from status inner join task on status.id=task.status_id;
+
+-- get the name of each status, along with a count of how many tasks have the status
+select count(task.id), name from  status join task on status.id=task.status_id group by status.name;
+
+-- get the name of all status, sorted by the status with most task first.
+select count(task.id), name from  task inner join status on status.id=task.status_id group by status.name order by task.id desc;
+
+-- second week
+-- Add a task
+insert into task ( title, description, created, updated, due_date, status_id, user_id) values ('Car Cleen', 'how to wash the car.', '2017-10-25 06:54:16', '2017-10-15 13:05:09', null, 2, 1);
+
+-- change the title of the task
+update task set title = 'description to car wash' where user_id = 1;
```

Comment:
looks good.
Try to make it a little bit advance.

change part of the title
find the record with other column names except 'id'